core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1379,14 +1374,7 @@ else if (isAggregation) {
                 throw new TrinoException(e::getErrorCode, extractLocation(node), e.getMessage(), e);
             }
 
-            if (function.getSignature().getName().equals(ARRAY_CONSTRUCTOR_NAME)) {
```

Comment:
We might want some general catch.  If the list is not constant, I expect there is a limit between 5-10 k given the number of bytecodes to push data on the stack and to call the methods.  If it is constant, I expect there is a point where it it just too much data, althought they likely hit the 1m query limit first.

Reply:
It’s caught by the expression compiler, just like with any other large expression.